### PR TITLE
enhance: use the key lock and concurrent map in the msg dispatcher client

### PR DESCRIPTION
**Pull request by SimFG**
/kind improvement

### CODE REVIEW

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

File: internal/util/flowgraph/node.go [hunk 1]
```diff
@@ -92,8 +92,8 @@ func (nodeCtxManager *nodeCtxManager) workNodeStart() {
 			name := fmt.Sprintf("nodeCtxTtChecker-%s", curNode.node.Name())
 			checker = timerecord.NewChecker(name, manager)
 			curNode = curNode.downstream
-			defer checker.Close()
 		}
+		defer checker.Close()
```
Comment (bigsheeper) [inline internal/util/flowgraph/node.go:96]: don't close here, see: https://github.com/milvus-io/milvus/pull/34339

**Pull request merged**